experimenter/experimenter/features/manifests/nimbus_fml_loader.py

```diff
@@ -1,40 +1,58 @@
-import logging
 import os
 
 import yaml
+from packaging import version
 
 from experimenter.settings import BASE_DIR
 
-# Todo: Connect FML https://mozilla-hub.atlassian.net/browse/EXP-3791
+# Todo: Connect FML https://mozilla-hub.atlassian.net/browse/EXP-3845
 # from nimbus-experimenter import Fml
 
-logger = logging.getLogger()
-
 
 class NimbusFmlLoader:
-    def __init__(self, application: str, channel: str):
+    __base_path = os.path.join(BASE_DIR, "features", "manifests", "apps.yaml")
+
+    def __init__(self, application: str, channel: str, file_location=__base_path):
         self.application: str = application
         self.channel: str = channel
-
-        path = os.path.join(BASE_DIR, "features", "manifests", "apps.yaml")
-        fml_local = ""
-        if os.path.exists(path):
-            with open(path) as application_yaml_file:
-                application_data = yaml.load(
-                    application_yaml_file.read(), Loader=yaml.Loader
-                )
-                for feature_slug in application_data:
-                    if feature_slug in application:
-                        fml_local = application_data[feature_slug]["fml_path"]
-                        logger.info(str(fml_local))
-                        break
-        if fml_local != "":
-            self.fml_loader = self.create(path, channel)
-            logger.info("FML loader created")
-        else:
-            logger.error("Failed to find fml path for application: " + application)
-
-    def create(self, path: str, channel: str):
+        self.application_data = self.get_application_data(application, file_location)
+
+    @staticmethod
+    def get_application_data(app, file_location=__base_path):
+        if os.path.exists(file_location):
+            with open(file_location) as application_yaml_file:
+                file = yaml.load(application_yaml_file.read(), Loader=yaml.Loader)
+                for app_name in file:
+                    if app_name in app:
+                        return file[app_name]
```

Comment:
Just a thought, rather than touch disk lazily like this, we could just load all these files into memory at startup and then do dict lookups when data is requested.

Reply:
I think that's the way we want to go eventually